src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -241,6 +240,30 @@ internal static bool TryGetApiVersion(Uri startRequestUri, out ReadOnlySpan<char
                     return _startRequestUri.AbsoluteUri;
             }
 
+            // If body contains resourceLocation, use it: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#target-resource-location
+            var contentStream = response.ContentStream;
+            if (contentStream is { CanSeek: true, Length: > 0 })
+            {
+                try
+                {
+                    using JsonDocument document = JsonDocument.Parse(contentStream);
```

Comment:
We are parsing in IsFinalState and in this method now.  Can we avoid the double parse?

Reply:
Fixed